### ISSUE TITLE
getLastTransactionID() returns a tailing carriage return (ASCII code 13) on Postfix

### ISSUE THREAD

**Issue by colandchan** [PHPMailer/PHPMailer#1247]
# Problem description
A trailing carriage return (ASCII code 13) is added into getLastTransactionID() on Postfix.
For example if the queue ID is "AAAAAAAAAA" then getLastTransactionID() will returns "AAAAAAAAAA\n"

Software version:
PHPMailer 6.0.1
CentOS 7.4.1708
Postfix version 2.10.1	

# Code to reproduce
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\Exception;

require __DIR__."/vendor/autoload.php";

$mail = new PHPMailer(true);

$mail->CharSet="UTF-8";
$mail->isSMTP();
$mail->SMTPDebug = 4;
$mail->Host = "127.0.0.1";
$mail->Port = "25";
$mail->SMTPAuth = false;
$mail->SMTPSecure = "tls";
$mail->setFrom("coland@*****MASKED*****");
$mail->addAddress("coland@*****MASKED*****");
$mail->SMTPOptions = array(
	"ssl"=>array(
		"allow_self_signed" => true,
		"verify_peer"  => false,
		"verify_peer_name"  => false,
	)
);

$mail->isHTML(true);
$mail->MsgHTML("Bug proof");
$mail->Subject = "Bug proof";

if($mail->send()){
echo "~~~~~DEBUG START ~~~~~\n";
$getLastTransactionID=$mail->getSMTPInstance()->getLastTransactionID();
var_dump($getLastTransactionID);
for($i=0; $i<strlen($getLastTransactionID); $i++){
echo "Character (".$i."): ".ord($getLastTransactionID[$i])."=>".$getLastTransactionID[$i]."\n";
}
echo "~~~~~DEBUG END ~~~~~\n";
}

# Debug output
2017-11-19 03:24:39	Connection: opening to 127.0.0.1:25, timeout=300, options=array (
                   	                    'ssl' => 
                   	                    array (
                   	                      'allow_self_signed' => true,
                   	                      'verify_peer' => false,
                   	                      'verify_peer_name' => false,
                   	                    ),
                   	                  )
2017-11-19 03:24:39	Connection: opened
2017-11-19 03:24:39	SMTP INBOUND: "220 *****MASKED***** ESMTP Postfix"
2017-11-19 03:24:39	SERVER -> CLIENT: 220 *****MASKED***** ESMTP Postfix
2017-11-19 03:24:39	CLIENT -> SERVER: EHLO *****MASKED*****.ap-southeast-1.compute.amazonaws.com
2017-11-19 03:24:39	SMTP INBOUND: "250-*****MASKED*****"
2017-11-19 03:24:39	SMTP INBOUND: "250-PIPELINING"
2017-11-19 03:24:39	SMTP INBOUND: "250-SIZE 10240000"
2017-11-19 03:24:39	SMTP INBOUND: "250-VRFY"
2017-11-19 03:24:39	SMTP INBOUND: "250-ETRN"
2017-11-19 03:24:39	SMTP INBOUND: "250-STARTTLS"
2017-11-19 03:24:39	SMTP INBOUND: "250-AUTH PLAIN LOGIN"
2017-11-19 03:24:39	SMTP INBOUND: "250-ENHANCEDSTATUSCODES"
2017-11-19 03:24:39	SMTP INBOUND: "250-8BITMIME"
2017-11-19 03:24:39	SMTP INBOUND: "250 DSN"
2017-11-19 03:24:39	SERVER -> CLIENT: 250-*****MASKED*****
                   	                  250-PIPELINING
                   	                  250-SIZE 10240000
                   	                  250-VRFY
                   	                  250-ETRN
                   	                  250-STARTTLS
                   	                  250-AUTH PLAIN LOGIN
                   	                  250-ENHANCEDSTATUSCODES
                   	                  250-8BITMIME
                   	                  250 DSN
2017-11-19 03:24:39	CLIENT -> SERVER: STARTTLS
2017-11-19 03:24:39	SMTP INBOUND: "220 2.0.0 Ready to start TLS"
2017-11-19 03:24:39	SERVER -> CLIENT: 220 2.0.0 Ready to start TLS
2017-11-19 03:24:39	CLIENT -> SERVER: EHLO *****MASKED*****.ap-southeast-1.compute.amazonaws.com
2017-11-19 03:24:39	SMTP INBOUND: "250-*****MASKED*****"
2017-11-19 03:24:39	SMTP INBOUND: "250-PIPELINING"
2017-11-19 03:24:39	SMTP INBOUND: "250-SIZE 10240000"
2017-11-19 03:24:39	SMTP INBOUND: "250-VRFY"
2017-11-19 03:24:39	SMTP INBOUND: "250-ETRN"
2017-11-19 03:24:39	SMTP INBOUND: "250-AUTH PLAIN LOGIN"
2017-11-19 03:24:39	SMTP INBOUND: "250-ENHANCEDSTATUSCODES"
2017-11-19 03:24:39	SMTP INBOUND: "250-8BITMIME"
2017-11-19 03:24:39	SMTP INBOUND: "250 DSN"
2017-11-19 03:24:39	SERVER -> CLIENT: 250-*****MASKED*****
                   	                  250-PIPELINING
                   	                  250-SIZE 10240000
                   	                  250-VRFY
                   	                  250-ETRN
                   	                  250-AUTH PLAIN LOGIN
                   	                  250-ENHANCEDSTATUSCODES
                   	                  250-8BITMIME
                   	                  250 DSN
2017-11-19 03:24:39	CLIENT -> SERVER: MAIL FROM:<coland@*****MASKED*****>
2017-11-19 03:24:39	SMTP INBOUND: "250 2.1.0 Ok"
2017-11-19 03:24:39	SERVER -> CLIENT: 250 2.1.0 Ok
2017-11-19 03:24:39	CLIENT -> SERVER: RCPT TO:<coland@*****MASKED*****>
2017-11-19 03:24:39	SMTP INBOUND: "250 2.1.5 Ok"
2017-11-19 03:24:39	SERVER -> CLIENT: 250 2.1.5 Ok
2017-11-19 03:24:39	CLIENT -> SERVER: DATA
2017-11-19 03:24:39	SMTP INBOUND: "354 End data with <CR><LF>.<CR><LF>"
2017-11-19 03:24:39	SERVER -> CLIENT: 354 End data with <CR><LF>.<CR><LF>
2017-11-19 03:24:39	CLIENT -> SERVER: Date: Sun, 19 Nov 2017 11:24:39 +0800
2017-11-19 03:24:39	CLIENT -> SERVER: To: coland@*****MASKED*****
2017-11-19 03:24:39	CLIENT -> SERVER: From: coland@*****MASKED*****
2017-11-19 03:24:39	CLIENT -> SERVER: Subject: Bug proof
2017-11-19 03:24:39	CLIENT -> SERVER: Message-ID: <yqZlk4nVtHw3zlbpT7KBfrM1l1lzz22P4GhsvbocyMI@*****MASKED*****.ap-southeast-1.compute.amazonaws.com>
2017-11-19 03:24:39	CLIENT -> SERVER: X-Mailer: PHPMailer 6.0.1 (https://github.com/PHPMailer/PHPMailer)
2017-11-19 03:24:39	CLIENT -> SERVER: MIME-Version: 1.0
2017-11-19 03:24:39	CLIENT -> SERVER: Content-Type: multipart/alternative;
2017-11-19 03:24:39	CLIENT -> SERVER: 	boundary="b1_yqZlk4nVtHw3zlbpT7KBfrM1l1lzz22P4GhsvbocyMI"
2017-11-19 03:24:39	CLIENT -> SERVER: Content-Transfer-Encoding: 8bit
2017-11-19 03:24:39	CLIENT -> SERVER:
2017-11-19 03:24:39	CLIENT -> SERVER: This is a multi-part message in MIME format.
2017-11-19 03:24:39	CLIENT -> SERVER: --b1_yqZlk4nVtHw3zlbpT7KBfrM1l1lzz22P4GhsvbocyMI
2017-11-19 03:24:39	CLIENT -> SERVER: Content-Type: text/plain; charset=us-ascii
2017-11-19 03:24:39	CLIENT -> SERVER:
2017-11-19 03:24:39	CLIENT -> SERVER: Bug proof
2017-11-19 03:24:39	CLIENT -> SERVER:
2017-11-19 03:24:39	CLIENT -> SERVER: --b1_yqZlk4nVtHw3zlbpT7KBfrM1l1lzz22P4GhsvbocyMI
2017-11-19 03:24:39	CLIENT -> SERVER: Content-Type: text/html; charset=us-ascii
2017-11-19 03:24:39	CLIENT -> SERVER:
2017-11-19 03:24:39	CLIENT -> SERVER: Bug proof
2017-11-19 03:24:39	CLIENT -> SERVER:
2017-11-19 03:24:39	CLIENT -> SERVER:
2017-11-19 03:24:39	CLIENT -> SERVER: --b1_yqZlk4nVtHw3zlbpT7KBfrM1l1lzz22P4GhsvbocyMI--
2017-11-19 03:24:39	CLIENT -> SERVER:
2017-11-19 03:24:39	CLIENT -> SERVER: .
2017-11-19 03:24:39	SMTP INBOUND: "250 2.0.0 Ok: queued as 2AB27CF2888"
2017-11-19 03:24:39	SERVER -> CLIENT: 250 2.0.0 Ok: queued as 2AB27CF2888
2017-11-19 03:24:39	CLIENT -> SERVER: QUIT
2017-11-19 03:24:39	SMTP INBOUND: "221 2.0.0 Bye"
2017-11-19 03:24:39	SERVER -> CLIENT: 221 2.0.0 Bye
2017-11-19 03:24:39	Connection: closed
~~~~~DEBUG START ~~~~~
string(12) "2AB27CF2888
"
Character (0): 50=>2
Character (1): 65=>A
Character (2): 66=>B
Character (3): 50=>2
Character (4): 55=>7
Character (5): 67=>C
Character (6): 70=>F
Character (7): 50=>2
Character (8): 56=>8
Character (9): 56=>8
Character (10): 56=>8
Character (11): 13=>
~~~~~DEBUG END ~~~~~

**Comment by Synchro** [comment]
That is strange. This is the (only) line that sets the transaction ID:

    $this->last_smtp_transaction_id = trim($matches[1]);

`trim()` strips ` \t\n\r\0\x0B` by default ([docs](http://php.net/trim)). So I don't know how that CR is getting through. I'll see if I can reproduce it.

**Comment by colandchan** [comment]
It is `$this->last_smtp_transaction_id = $matches[1];` on 6.0.1,
but `$this->last_smtp_transaction_id = trim($matches[1]);` on master

**Comment by Synchro** [comment]
Ah, ok well it will be in the next release then. In the mean time you can use dev-master.

**Comment by Synchro** [comment]
This fix has been released in 6.0.2.
